### PR TITLE
Feature/feature group subscribe

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/community/Group.java  10
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/ForumsFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupFragment.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriptionComponentView.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddSubscription.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriptionView.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupSubscriptionsFragment.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddSubscription.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscribersView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriberComponentView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupFragmentAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/CommunityFragmentAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/CommunityFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/UserSubscriptionsFragment.java  2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriptionComponentView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5387123&bid=17662317)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/CommunityFragment.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            switch (position) {
+                case CommunityFragmentAdapter.GROUPS_FRAGMENT:
+                    tab.setText(R.string.community_group_search);
+                    break;
+                case CommunityFragmentAdapter.USER_SUBSCRIPTIONS_FRAGMENT:
+                    tab.setText(R.string.community_my_subscriptions);
+                    break;
+                default:
+            }
+            });
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/CommunityFragment.java:63]: Codacy Issue found: ['block rcurly' has incorrect indentation level 12, expected level should be 8.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5387123)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupFragment.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            switch (position) {
+                case InfoGroupFragmentAdapter.FORUMS_FRAGMENT:
+                    tab.setText(R.string.info_group_forums);
+                    break;
+                case InfoGroupFragmentAdapter.GROUP_SUBSCRIPTIONS_FRAGMENT:
+                    tab.setText(R.string.info_group_subscriptors);
+                    break;
+                default:
+            }
+            });
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupFragment.java:87]: Codacy Issue found: ['block rcurly' has incorrect indentation level 12, expected level should be 8.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5387123)

**Review by enrichp — APPROVED** [review]

**Review by xavier-campos-diaz — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscribersView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriptionView.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrAddSubscription.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/CommunityFragmentAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/CommunityFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupFragmentAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/ForumsFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/UserSubscriptionsFragment.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupFragment.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriptionComponentView.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrAddSubscription.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/InfoGroupSubscriptionsFragment.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriberComponentView.java  2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/GroupComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/SubscriptionComponentView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5387123&bid=17662317)

**Pull request merged**